testgrid/transfigure.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+
+# A script that uses configurator to create a YAML file, and then pushes that
+# YAML file to k8s/test-infra's config/testgrids directory
+#
+# Used by teams that host different instances of Prow, but want to show their
+# results on testgrid.k8s.io
+#
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+main() {
+  branch="transfigure-branch"
+
+  if [[ $# -lt 1 ]]; then
+      echo "Usage: $(basename "$0") </path/to/github/token> [prow_config] [prow-job_config] [testgrid-yaml] [subdir-path]"
+      echo "Path to GitHub token required"
+      exit 1
+  fi
+
+  token=$(readlink -f $1)
+  shift
+  user-from-token
+  ensure-git-config
+  parse-args "$@"
+
+  echo "Getting kubernetes/test-infra repo"
+  git clone https://github.com/kubernetes/test-infra.git || echo "Found test-infra directory; proceeding"
+  k8srepo=$(readlink -f test-infra)
+  target=$(readlink -f test-infra/config/testgrids)
+  trap "cleanup-repository" ERR
+  echo "Created repo at $k8srepo"
+
+  echo "Checking out ${branch}"
+  cd $k8srepo
+  git checkout -B ${branch}
+
+  echo "Generating testgrid yaml"
+  go run k8s.io/test-infra/testgrid/cmd/configurator \
+    --prow-config ${prow_config} \
+    --prow-job-config ${job_config} \
+    --output-yaml \
+    --yaml ${testgrid_config} \
+    --oneshot \
+    --output "${target}/${org_repo}/gen-config.yaml"
+
+  git add --all
+
+  if ! git diff --quiet ; then
+    echo "Transfigure did not change anything. Aborting no-op bump"
+    exit 0
+  fi
+
+  title="Update TestGrid for ${org_repo}"
+  git commit -m "${title}"
+  echo -e "Pushing commit to ${user}/${remote_fork_repo}:${branch}..." >&2
```

Comment:
nit
```suggestion
  echo "Pushing commit to ${user}/${remote_fork_repo}:${branch}..." >&2
```

Also, any particular reason you decided to redirect to stderr (i.e. `>&2`) here. Is this intentional?

Reply:
I'm honestly not sure under which circumstances I should redirect to stderr or stdout for bash scripts. Is there a style guide that I can look at? Should stderr be for "debug-level" messages only?

Reply:
I found this, but it's very brief
https://google.github.io/styleguide/shell.xml?showone=STDOUT_vs_STDERR#STDOUT_vs_STDERR

Reply:
Yes, `stderr` is typically used for either errors or debug messages that would not be piped to another command. Really, probably all the echos here could be redirected to `stderr` following that rule. Bash is like the wild west, so I'm not sure style guide. @fejta might have some good resources.